test/distributed/_shard/test_replicated_tensor.py

```diff
@@ -74,3 +77,143 @@ def test_replicated_tensor_inter_op_tensor(self):
+        self.assertEqual(new_tensor.local_tensor(), st.local_tensor() + 4)
+
+        # sub op
+        new_tensor = replica_tensor - st
+        self.assertTrue(
+            isinstance(new_tensor, sharded_tensor.ShardedTensor)
+            and not isinstance(new_tensor, ReplicatedTensor)
+        )
+        self.assertEqual(new_tensor.local_tensor(), 4 - st.local_tensor())
+        # test torch.sub
+        new_tensor = torch.sub(replica_tensor, st)
+        self.assertEqual(new_tensor.local_tensor(), 4 - st.local_tensor())
+
+        new_tensor = st - replica_tensor
+        self.assertEqual(new_tensor.local_tensor(), st.local_tensor() - 4)
+
+        # mul op
+        new_tensor = replica_tensor * st
+        self.assertTrue(
+            isinstance(new_tensor, sharded_tensor.ShardedTensor)
+            and not isinstance(new_tensor, ReplicatedTensor)
+        )
+        self.assertEqual(new_tensor.local_tensor(), 4 * st.local_tensor())
+        # test torch.mul
+        new_tensor = torch.mul(replica_tensor, st)
+        self.assertEqual(new_tensor.local_tensor(), 4 * st.local_tensor())
+
+        new_tensor = st * replica_tensor
+        self.assertEqual(new_tensor.local_tensor(), 4 * st.local_tensor())
+
+        # div op
+        new_tensor = replica_tensor / st
+        self.assertTrue(
+            isinstance(new_tensor, sharded_tensor.ShardedTensor)
+            and not isinstance(new_tensor, ReplicatedTensor)
+        )
+        self.assertEqual(new_tensor.local_tensor(), 4 / st.local_tensor())
+        # test torch.div
+        new_tensor = torch.div(replica_tensor, st)
+        self.assertEqual(new_tensor.local_tensor(), 4 / st.local_tensor())
+
+        new_tensor = st / replica_tensor
+        self.assertEqual(new_tensor.local_tensor(), st.local_tensor() / 4)
+
+    @with_comms(init_rpc=False)
+    @skip_if_lt_x_gpu(4)
+    @requires_nccl()
+    def test_replicated_tensor_implicit_broadcasting(self):
+        #  use same seed
+        torch.manual_seed(self.rank)
+
+        # test implicit broadcasting
+        local_tensor1 = torch.rand(12, 3, device=f"cuda:{self.rank}") * 4
+        local_tensor2 = torch.ones(1, 3, device=f"cuda:{self.rank}")
+
+        spec = ChunkShardingSpec(
+            dim=0,
+            placements=[
+                "rank:0/cuda:0",
+                "rank:1/cuda:1",
+                "rank:2/cuda:2",
+                "rank:3/cuda:3",
+            ],
+        )
+
+        st = _shard_tensor(local_tensor1, spec, src_rank=0)
+        replica_tensor = ReplicatedTensor(local_tensor2)
+
+        # replicated tensor should automatically broadcasted
+        res = torch.add(st, replica_tensor)
+
+        self.assertIsInstance(res, sharded_tensor.ShardedTensor)
+        output = torch.empty((12, 3)) if self.rank == 0 else None
+        res.gather(dst=0, out=output)
+
+        if self.rank == 0:
+            local_output = torch.add(local_tensor1, local_tensor2)
+            self.assertEqual(output, local_output)
```

Comment:
It looks like we are doing this local vs sharded + replicated verification only here for torch.add. Shouldn't we run `test_basic_math_ops` with ShardedTensor op ReplicatedTensor and verify it is on par with local compute?